src/ui/backend/canvas/animate.js

```diff
@@ -178,33 +285,39 @@ exports.easeInOutBounce   = 34;
 
 function getTransition(n) {
 	return (typeof n == 'function' ? n : TRANSITIONS[n | 0]);
-}
+};
 
 var Frame = Class(function () {
 	this.init = function (opts) {
 		this.subject = opts.subject;
 		this.target = opts.target;
-		this.duration = opts.duration || 0;
+		this.duration = opts.duration === 0 ? 0 : (opts.duration || 500);
 		this.transition = getTransition(opts.transition);
 		this.onTick = opts.onTick;
 	};
 
 	this.exec = function () {};
-	this.onInterrupt = function () {};
+	this.debugLog = function () {};
 });
 
 var CallbackFrame = Class(Frame, function () {
+	this.init = function (opts) {
+		Frame.prototype.init.call(this, opts);
+		// CallbackFrames act like tick functions when given durations
+		this.duration = opts.duration || 0;
+	};
+
 	this.exec = function (tt, t) {
 		this.target.apply(this.subject, arguments);
 	};
 });
 
-// a wait frame is just a frame that does nothing... so we
-// don't need to do anything!
+// a wait frame is just a frame that does nothing... so don't do anything!
 var WaitFrame = Frame;
```

Comment:
If `WaitFrame == Frame` we should probably just remove `WaitFrame`, since it might just cause confusion (now for all frames, `frame instanceof WaitFrame == true`)